python_modules/dagster/dagster/_core/definitions/assets.py

```diff
@@ -1683,39 +1687,35 @@ def _asset_specs_from_attr_key_params(
                     # Value here is irrelevant, because it will be replaced by value from
                     # NodeDefinition
                     skippable=False,
+                    # Value here is irrelevant, because it will be replaced by value from partitions_def
+                    partitions_def=None,
                 )
             )
 
     return result
 
 
-def _validate_self_deps(
-    input_keys: Iterable[AssetKey],
-    output_keys: Iterable[AssetKey],
-    partition_mappings: Mapping[AssetKey, PartitionMapping],
-    partitions_def: Optional[PartitionsDefinition],
-) -> None:
-    output_keys_set = set(output_keys)
-    for input_key in input_keys:
-        if input_key in output_keys_set:
-            if input_key in partition_mappings:
-                partition_mapping = partition_mappings[input_key]
-                time_window_partition_mapping = get_self_dep_time_window_partition_mapping(
-                    partition_mapping, partitions_def
+def _validate_self_deps(specs: Iterable[AssetSpec]) -> None:
```

Comment:
This changes this check to entirely operate on the deps in the specs, instead of going through inputs and outputs